pubsub/api/PubsubTest.cs

```diff
@@ -0,0 +1,444 @@
+                }
+                System.Threading.Thread.Sleep(delayMs);
+                delayMs *= 2;  // Exponential back-off.
+            }
+            throw new AggregateException(exceptions);
+        }
+
+        private void RetryRpc(Action action)
+        {
+            try {
+                RetryRpc(() => { action(); return 0; });
+            }
+            catch (AggregateException exceptions)
+            {
+                throw exceptions;
+            }
+        }
+        // [END retry]
+
+        public void RpcRetry(string topicId, string subscriptionId,
+            PublisherClient publisher, SubscriberClient subscriber)
+        {
+            string topicName = PublisherClient.FormatTopicName(_projectId, topicId);
+            string subscriptionName =
+                SubscriberClient.FormatSubscriptionName(_projectId, subscriptionId);
+            try
+            {
+                RetryRpc(() =>
+                {
+                    // Subscribe to Topic
+                    // This should fail since Topic has not yet been created
+                    subscriber.CreateSubscription(subscriptionName, topicName,
+                        pushConfig: null, ackDeadlineSeconds: 60);
+                });
+            }
+            catch (AggregateException exceptions)
+            {
+                if (exceptions.InnerExceptions.Count() == _retryCount)
+                {
+                    // [START retry]
+                    try
+                    {
+                        RetryRpc(() =>
+                        {
+                            //Create Topic
+                            publisher.CreateTopic(topicName);
+                        });
+                    }
+                    catch (RpcException e) when 
+                        (e.Status.StatusCode != StatusCode.AlreadyExists)
+                    {
+                        // A StatusCode of "AlreadyExists" is ok.  
+                        // It means the Topic already exists. 
+                        // Otherwise throw exception.
+                        throw;
+                    }
+                    // [END retry]
+                    try
+                    {
+                        RetryRpc(() =>
+                        {
+                            // Subscribe to Topic
+                            subscriber.CreateSubscription(subscriptionName, topicName,
+                                pushConfig: null, ackDeadlineSeconds: 60);
+                        });
+                    }
+                    catch (RpcException e) when
+                        (e.Status.StatusCode != StatusCode.AlreadyExists)
+                    {
+                        // A StatusCode of "AlreadyExists" is ok.  
+                        // It means the Subscription already exists. 
+                        // Otherwise throw exception.
+                        throw;
+                    }
+                }
+            }
+        }

```

Comment:
In-lining the exception handling code might make it even longer b/c currently RetryRpc() allows CreateTopic and CreateSubscription to share the RPC looping/retry functionality. I'll try out in-lining to see if something more succinct is possible while still demonstrating RPC retries (which is a required code sample section in the current documentation).